docker/install/ubuntu_install_llvm_from_source.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# This script builds LLVM and clang from the llvm-project tarball
+# using CMake. It is tested with LLVM from version 15.
+
+set -e
+
+LLVM_VERSION_MAJOR=$1
+
+detect_llvm_version() {
+  curl -sL "https://api.github.com/repos/llvm/llvm-project/releases?per_page=100" | \
+    grep tag_name | \
+    grep -o "llvmorg-${LLVM_VERSION_MAJOR}[^\"]*" | \
+    grep -v rc | \
+    sed -e "s/^llvmorg-//g" | \
+    head -n 1
+}
+
+LLVM_VERSION=$(detect_llvm_version)
+echo ${LLVM_VERSION}
+
+curl -sL \
+  https://github.com/llvm/llvm-project/releases/download/llvmorg-${LLVM_VERSION}/llvm-project-${LLVM_VERSION}.src.tar.xz \
+  -o llvm-project-${LLVM_VERSION}.src.tar.xz
+unxz llvm-project-${LLVM_VERSION}.src.tar.xz
+tar xf llvm-project-${LLVM_VERSION}.src.tar
```

Comment:
Does this not work? Surprised `tar` doesn't sniff the format.

```suggestion
tar xf llvm-project-${LLVM_VERSION}.src.tar.xz
```